pkg/controller/syncset/syncset_controller.go

```diff
@@ -0,0 +1,439 @@
+	h := r.applierBuilder(kubeConfig, r.logger)
+
+	for _, resource := range ss.Spec.Resources {
+		info, err := h.Info(resource.Raw)
+		if err != nil {
+			syncSetStatus.Conditions = r.setUnknownObjectSyncCondition(syncSetStatus.Conditions, err)
+			return err
+		}
+
+		resourceHash := md5.Sum(resource.Raw)
+		resourceSyncStatus := hivev1.SyncStatus{
+			APIVersion: info.APIVersion,
+			Kind:       info.Kind,
+			Name:       info.Name,
+			Namespace:  info.Namespace,
+			Hash:       fmt.Sprintf("%x", resourceHash[:]),
+		}
+
+		// determine if resource has been created or is different based on the resource md5 sum
+		found := false
+		different := false
+		for _, rss := range syncSetStatus.Resources {
+			if rss.Name == resourceSyncStatus.Name && rss.Namespace == resourceSyncStatus.Namespace {
+				found = true
+				if rss.Hash != resourceSyncStatus.Hash {
+					different = true
+				}
+				break
+			}
+		}
+
+		if !found || different {
+			ssLog.Debugf("resource %v has not been created or is different", info.Name)
+			err = h.Apply(resource.Raw)
+			if err != nil {
+				ssLog.WithError(err).Error("unable to apply resource")
+			}
+			resourceSyncStatus.Conditions = r.setApplySuccessSyncCondition(resourceSyncStatus.Conditions, err)
+			syncSetStatus.Resources = appendOrUpdateSyncStatus(syncSetStatus.Resources, resourceSyncStatus)
+		}
+	}
+
+	syncSetStatus.Conditions = r.setUnknownObjectSyncCondition(syncSetStatus.Conditions, nil)
+	return nil
+}
+
+func appendOrUpdateSyncSetObjectStatus(statusList []hivev1.SyncSetObjectStatus, syncSetObjectStatus hivev1.SyncSetObjectStatus) []hivev1.SyncSetObjectStatus {
+	found := false
+	for _, ssos := range statusList {
+		if ssos.Name == syncSetObjectStatus.Name {
+			found = true
+			ssos = syncSetObjectStatus
+			break
+		}
+	}
+	if !found {
+		statusList = append(statusList, syncSetObjectStatus)
+	}
+	return statusList
+}
+
+func appendOrUpdateSyncStatus(statusList []hivev1.SyncStatus, syncStatus hivev1.SyncStatus) []hivev1.SyncStatus {
+	found := false
+	for i, ss := range statusList {
+		if ss.Name == syncStatus.Name && ss.Namespace == syncStatus.Namespace && ss.Kind == syncStatus.Kind {
+			found = true
+			statusList[i] = syncStatus
+			break
+		}
+	}
+	if !found {
+		statusList = append(statusList, syncStatus)
+	}
+	return statusList
+}
+
+func findSyncSetStatus(name string, cdStatus hivev1.ClusterDeploymentStatus) hivev1.SyncSetObjectStatus {
+	syncSetStatus := hivev1.SyncSetObjectStatus{Name: name}
+	for _, ssos := range cdStatus.SyncSetStatus {
+		if name == ssos.Name {
+			syncSetStatus = ssos
+			break
+		}
+	}
+	return syncSetStatus
+}
+
+func (r *ReconcileSyncSet) updateClusterDeploymentStatus(cd *hivev1.ClusterDeployment, origCD *hivev1.ClusterDeployment, cdLog log.FieldLogger) error {
+	// Update cluster deployment status if changed:
+	if !reflect.DeepEqual(cd.Status, origCD.Status) {
+		cdLog.Infof("status has changed, updating cluster deployment")
+		cdLog.Debugf("orig: %v", origCD.Status)
```

Comment:
If we want to see the contents, we should at least output the serialized JSON, otherwise, it's not that useful.